### PR TITLE
Add `preferredmarkers` linter

### PR DESCRIPTION
**Pull request by saschagrunert**
Add a new linter that allows projects to enforce using preferred markers instead of equivalent but different marker identifiers.

- Marker preferences are configurable
- Auto-fix support
- Maintains marker expressions (e.g., :=value, :key=value) during replacement
- When multiple equivalent markers exist on the same field/type, consolidates them into a single preferred marker
- Handles markers inherited from type aliases
- Sorted processing ensures consistent, reproducible results

The linter is disabled by default and requires explicit configuration to enable.

Fixes https://github.com/kubernetes-sigs/kube-api-linter/issues/139

### CODE REVIEW

File: pkg/analysis/preferredmarkers/testdata/src/a/eof.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package a
+
+// Type at end of file without trailing newline
+// +kubebuilder:validation:Optional
+type EndOfFileType string // want `type EndOfFileType uses marker "kubebuilder:validation:Optional", should use preferred marker "k8s:optional" instead`
```
Comment (JoelSpeed) [inline pkg/analysis/preferredmarkers/testdata/src/a/eof.go]: Nit, is this deliberately missing the new line char?

File: pkg/analysis/preferredmarkers/analyzer.go [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+// formatExpressions formats marker expressions as a string suffix.
+// Returns ":=value" for unnamed expressions or ":key1=val1,key2=val2" for named expressions.
+func formatExpressions(expressions map[string]string) string {
+	if len(expressions) == 0 {
+		return ""
+	}
+
+	// Check for unnamed expression (simple case like "+marker:=value")
+	if unnamedValue, ok := expressions[markers.UnnamedExpression]; ok {
+		return ":=" + unnamedValue
```
Comment (JoelSpeed) [inline pkg/analysis/preferredmarkers/analyzer.go]: I don't think `:=` is universally compatible, just use `=`.

Kubebuilder supports `:=`, but to my knowledge, the `+k8s` markers do not

File: pkg/analysis/preferredmarkers/testdata/src/a/eof.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (saschagrunert) [reply pkg/analysis/preferredmarkers/testdata/src/a/eof.go]: Yeah I think those tests are not that helpful for edge cases like this. I decided to remove them again.

File: pkg/analysis/preferredmarkers/testdata/src/a/a.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+// +custom:optional
+type MultipleEquivalentMarkersType string // want `type MultipleEquivalentMarkersType uses markers "custom:optional", "kubebuilder:validation:Optional", should use preferred marker "k8s:optional" instead`
+
+type Test struct {
+	// Field with kubebuilder optional marker (should be k8s:optional)
+	// +kubebuilder:validation:Optional
+	OptionalFieldKubebuilder string `json:"optionalFieldKubebuilder"` // want `field OptionalFieldKubebuilder uses marker "kubebuilder:validation:Optional", should use preferred marker "k8s:optional" instead`
+
+	// Inherited from type alias
+	OptionalFieldKubebuilderTypeAlias OptionalTypeKubebuilder `json:"optionalFieldKubebuilderTypeAlias"` // want `field OptionalFieldKubebuilderTypeAlias uses marker "kubebuilder:validation:Optional", should use preferred marker "k8s:optional" instead`
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply pkg/analysis/preferredmarkers/testdata/src/a/a.go]: I changed the field checking to only report markers directly on fields, not inherited ones.

File: pkg/analysis/preferredmarkers/initializer_test.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+			Expect(ok).To(BeTrue())
+
+			errs := ci.ValidateConfig(in.config, field.NewPath("preferredmarkers"))
+			if len(in.expectedErr) > 0 {
+				Expect(errs.ToAggregate()).To(MatchError(in.expectedErr))
+			} else {
+				Expect(errs).To(HaveLen(0), "No errors were expected")
+			}
+		},
+			Entry("With a valid preferredmarkers configuration", testCase{
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply pkg/analysis/preferredmarkers/initializer_test.go]: I added a test case in initializer_test.go
